Mathlib/NumberTheory/FLT/Three.lean

```diff
@@ -0,0 +1,68 @@
+/-
+Copyright (c) 2024 Riccardo Brasca. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Riccardo Brasca
+-/
+import Mathlib.Data.ZMod.Basic
+
+/-!
+# Fermat Last Theorem in the case `n = 3`
+The goal of this file is to prove Fermats Last theorem in the case `n = 3`.
+
+## Main results
+* `fermatLastTheoremThree_case_1`: the first case of Fermat Last Theorem when `n = 3`:
+  if `a b c : ℕ` are such that `¬ 3 ∣ a * b * c` then `a ^ 3 + b ^ 3 ≠ c ^ 3`.
+
+## TODO
+Prove case 2.
+-/
+
+open ZMod
+
+private lemma three_dvd_nine : 3 ∣ 9 := by norm_num
+
+private lemma mem_of_castHom_ne_zero {a : ZMod 9} (ha : castHom three_dvd_nine (ZMod 3) a ≠ 0) :
+    a ∈ ({1, 2, 4, 5, 7, 8} : Finset (ZMod 9)) := by
+  fin_cases a
+  · aesop
+  rotate_left 2
+  · exfalso
+    exact ha rfl
+  rotate_left 2
+  · exfalso
+    exact ha rfl
+  all_goals decide
+
+private lemma mem_of_coe_ne_zero {a : ℕ} (ha : (a : ZMod 3) ≠ 0) :
+    (a : ZMod 9) ∈ ({1, 2, 4, 5, 7, 8} : Finset (ZMod 9)) :=
+  mem_of_castHom_ne_zero (fun h ↦ ha <| by simpa using h)
+
+private lemma cube_mem_of_mem {a : ZMod 9} (ha : a ∈ ({1, 2, 4, 5, 7, 8} : Finset (ZMod 9))) :
+    a ^ 3 ∈ ({1, 8} : Finset (ZMod 9)) := by
+  fin_cases ha
+  all_goals decide
+
+private lemma fermatLastTheoremThree_ZMod9_case_1 {a b c : ZMod 9}
+  (ha : a ∈ ({1, 2, 4, 5, 7, 8} : Finset (ZMod 9)))
+  (hb : b ∈ ({1, 2, 4, 5, 7, 8} : Finset (ZMod 9)))
+  (hc : c ∈ ({1, 2, 4, 5, 7, 8} : Finset (ZMod 9))) :
+    a ^ 3 + b ^ 3 ≠ c ^ 3 := by
+  replace ha := cube_mem_of_mem ha
+  replace hb := cube_mem_of_mem hb
+  replace hc := cube_mem_of_mem hc
+  simp only [Finset.mem_insert, Finset.mem_singleton] at ha hb hc
+  rcases ha with (ha | ha) <;> rcases hb with (hb | hb) <;> rcases hc with (hc | hc)
+  all_goals simp only [ha, hb, hc]; decide
+
+/--If `a b c : ℕ` are such that `¬ 3 ∣ a * b * c` then `a ^ 3 + b ^ 3 ≠ c ^ 3`. -/
+theorem fermatLastTheoremThree_case_1 :
+    ∀ a b c : ℕ, ¬ 3 ∣ a * b * c → a ^ 3 + b ^ 3 ≠ c ^ 3 := by
```

Comment:
Why did you put the quantification right of the `:`?

Reply:
I just copied the statement in [FermatLastTheorem](https://leanprover-community.github.io/mathlib4_docs/Mathlib/NumberTheory/FLT/Basic.html#FermatLastTheorem), but it's probably better to follow the usual style.